content/docs/user-guide/programming/gems/code-gems.md

```diff
@@ -0,0 +1,43 @@
+---
+linkTitle: Code Gems
+title: Code Gem Specifications
+description: An overview of Code Gems in Open 3D Engine.
+weight: 300
+---
+
+**Open 3D Engine (O3DE)** code Gems contains code that let's you extend the O3DE Editor or integrate features and gameplay logic for your project. They can also contain assets that your code interacts with.
+
+## Creating a Gem
+
+Follow [Create an O3DE Gem](/docs/user-guide/programming/gems/creating) to create a Gem based on the **DefaultGem template**. Code and asset Gems have the same Gem directory structure and you can use the same command to create them. 
+
+
+## Gem code
+
+The code components of a Gem are located in the `Code` directory. The DefaultGem template provides the following boilerplate code for the Gem's source code in the `Code/Source` directory of your Gem.
+
+![An image of the file directory of Gem code](/images/user-guide/programming/gems/defaultgem-template-directory-code-source.png)
+
+You can view the Gem's source code from a Gem that you created or from the DefaultGem template that's located in the `templates/DefaultGem` directory of your engine source code.
+
+The following files are the code components of a Gem. 
+
+### `<Gem>ModuleInterface.h`, `<Gem>Module.cpp`
+
+The `<Gem>ModuleInterface` is the main Gem Module class that contains the entry point functions to connect the Gem's functionality to O3DE. The DefaultGem template already sets up the Gem Module to connect to O3DE by calling the `AZ_DECLARE_MODULE_CLASS` macro.
+
+For more information on modules and their responsibilities in O3DE, refer to [Gem Module System](/docs/user-guide/programming/gems/overview/).
+
+### `<Gem>SystemComponent.cpp`, `<Gem>SystemComponent.h`
+
+The `<Gem>SystemComponent` class is a global singleton class that's responsible for managing the Gem's code. The system components allows the component within the Gem Module to communicate with components from other Gems by connecting to their EBus. It also handles underlying tasks that govern the whole Gem such as initialization and shutdown, event handling, memory allocation, and connection to debuggers. The Gem Module registers the system component and the system component manages the Gem Module's life cycle: initialization, activativation, and deactivation. The system component activates when the Gem loads and deactivates when the Gem unloads.
```

Comment:
This is a lot of info in a short passage. Suggestion to make it easier to digest, maybe?

```suggestion
The `<Gem>SystemComponent` class is a global singleton class that's responsible for managing the Gem's code. The Gem Module registers the Gem System Component which manages the Gem Module's life cycle: initialization, activation, and deactivation. The Gem System Component activates when the Gem loads and deactivates when the Gem unloads.

The Gem System Component allows the component within the Gem Module to communicate with components from other Gems by connecting to their EBus. It also handles initialization and shutdown, events, memory allocation, and debugger connections. 
```